app/src/main/java/pm/gnosis/heimdall/common/di/modules/ApplicationModule.kt

```diff
@@ -76,18 +78,23 @@ class ApplicationModule {
 
     @Provides
     @Singleton
-    @Named(REST_CLIENT)
-    fun providesOkHttpClient(@Named(InterceptorsModule.REST_CLIENT_INTERCEPTORS) interceptors: @JvmSuppressWildcards List<Interceptor>): OkHttpClient {
-        return OkHttpClient.Builder().apply {
-            connectTimeout(10, TimeUnit.SECONDS)
-            readTimeout(10, TimeUnit.SECONDS)
-            writeTimeout(10, TimeUnit.SECONDS)
+    @Named(INFURA_REST_CLIENT)
+    fun providesInfuraOkHttpClient(okHttpClient: OkHttpClient, @Named(InterceptorsModule.REST_CLIENT_INTERCEPTORS) interceptors: @JvmSuppressWildcards List<Interceptor>): OkHttpClient {
+        return okHttpClient.newBuilder().apply {
```

Comment:
Why don't we check in the in the interceptor if the api key should be appended or not? We can change the endpoint URL via the settings, and the api key will still be appended.